--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1244,5 +1244,12 @@
   )
   TARGET_C_PROPERTIES (${HDF5_LIB_TARGET} STATIC)
+ 
+#======= PATCH: Without this, zlib.h is not found when configuring and building with Conan ============
+
+  target_link_libraries(${HDF5_LIB_TARGET} PRIVATE ZLIB::ZLIB)
+
+#======================================================================================================
+
   target_link_libraries (${HDF5_LIB_TARGET}
       PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS}
       PUBLIC "$<$<NOT:$<PLATFORM_ID:Windows>>:${CMAKE_DL_LIBS}>" "$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:MPI::MPI_C>"
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1294,5 +1294,12 @@
   )
   TARGET_C_PROPERTIES (${HDF5_LIBSH_TARGET} SHARED)
+ 
+#======= PATCH: Without this, zlib.h is not found when configuring and building with Conan ============
+
+  target_link_libraries(${HDF5_LIBSH_TARGET} PRIVATE ZLIB::ZLIB)
+
+#======================================================================================================
+
   target_link_libraries (${HDF5_LIBSH_TARGET}
       PRIVATE ${LINK_LIBS} ${LINK_COMP_LIBS}
               "$<$<OR:$<BOOL:${HDF5_ENABLE_THREADSAFE}>,$<BOOL:${HDF5_ENABLE_SUBFILING_VFD}>>:Threads::Threads>"
